app/models/piece.rb

```diff
@@ -109,4 +109,12 @@ def move_to!(x_new, y_new)
       piece.update_attributes(:position_x => x_destination, :position_y => y_destination)
     end
   end
+
+  def valid_move_vertical?(position_x, position_y)
+    true unless is_obstructed_vertically(position_x, position_y)
```

Comment:
nice using unless 😉.. 

what you could do here is cut it down to `!is_obstructed...` if it returns true, the valid_move_vertical will return false and vice versa..

Reply:
Ok great! I will do that and then resolve the conflicts and merge.